VisualPinball.Unity/VisualPinball.Unity.Editor/Layers/LayersHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace VisualPinball.Unity.Editor.Layers
+{
+	/// <summary>
+	/// This handler will construct a layer structure from the table loaded data.
+	/// It will then be in charge of layers management (add/remove/item assignation)
+	/// It will use the SceneVisibilityManager to synchronize layers/items visibility status & editor visibility
+	/// </summary>
+	class LayersHandler
+	{
+		public bool IsVisible(GameObject gameObj)
+		{
+			return !SceneVisibilityManager.instance.IsHidden(gameObj);
+		}
+
+		public void SetVisibility(GameObject gameObj, bool visibility, bool includeDescendants)
+		{
+			if (visibility) {
+				SceneVisibilityManager.instance.Show(gameObj, includeDescendants);
+			} else {
+				SceneVisibilityManager.instance.Hide(gameObj, includeDescendants);
+			}
+		}
+		public void SetVisibility(GameObject[] gameObjs, bool visibility, bool includeDescendants)
+		{
+			if (visibility) {
+				SceneVisibilityManager.instance.Show(gameObjs, includeDescendants);
+			} else {
+				SceneVisibilityManager.instance.Hide(gameObjs, includeDescendants);
+			}
+		}

```

Comment:
Seems unused?

Reply:
It is from treeview

Reply:
There are two overloads, the other `SetVisibility()` that takes in a single `GameObject` is used, I don't see usage of this one.

Reply:
oh yeah right, was preparing the multiselection support already.

Reply:
If you're sure we'll use it later, it's okay!